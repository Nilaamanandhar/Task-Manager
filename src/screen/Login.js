import React, {Component} from 'react';
import {
  View,
  Text,
  StyleSheet,
  ImageBackground,
  Image,
  TextInput,
  Button,
  TouchableHighlight,
  ActivityIndicator,
  Alert,
} from 'react-native';
import welcomeImage from '../assests/image/welcome.jpg';
export default class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      login: false,
      username: '',
      password: '',
      auth: [
        {username: 'developer', password: 'developer'},
        {username: 'webpoint', password: 'webpoint'},
      ],
    };
  }

  login = () => {
    if (this.state.username != '' && this.state.password != '') {
      let i = 0;
      this.state.auth.map((data, index) => {
        if (
          data.username == this.state.username &&
          data.password == this.state.password
        ) {
          this.props.navigation.navigate('main');
        } else {
          i++;
        }
      });
      if (this.state.auth.length === i) {
        Alert.alert('username or password is incorrect');
      }
    } else {
      Alert.alert('username or password is empty');
    }
    this.setState({login: false});
  };
  render() {
    return (
      <ImageBackground source={welcomeImage} style={{flex: 1}}>
        <View style={styles.loginContainer}>
          <Text style={styles.logintxt}>log in</Text>
          <View style={styles.loginFieldContainer}>
            <TextInput
              style={styles.loginField}
              value={this.state.username}
              onChangeText={username => this.setState({username})}
              placeholder="User Name"
              placeholderTextColor="white"
              underlineColorAndroid="transparent"></TextInput>
            <TextInput
              style={styles.loginField}
              value={this.state.password}
              secureTextEntry={true}
              onChangeText={password => this.setState({password})}
              placeholder="Password"
              placeholderTextColor="white"
              underlineColorAndroid="transparent"></TextInput>
            <TouchableHighlight onPress={this.login} style={styles.loginButton}>
              <Text style={styles.loginButtonText}>Login</Text>
            </TouchableHighlight>
          </View>
          <View>
            <Text style={styles.hint}>username: developer </Text>
            <Text style={styles.hint}>password: developer </Text>
          </View>
        </View>
      </ImageBackground>
    );
  }
}
const styles = StyleSheet.create({
  hint: {
    marginTop: 0,
    color: 'white',
    fontSize: 20,
  },
  loginContainer: {
    display: 'flex',
    flexDirection: 'column',
    flex: 1,
    justifyContent: 'space-between',
    alignItems: 'center',
    position: 'absolute',
    top: 40,
    left: 0,
    right: 0,
    zIndex: 10,
  },

  logintxt: {
    color: 'white',
    fontSize: 25,
    fontWeight: '500',
    marginTop: 15,
    opacity: 0.5,
  },
  loginField: {
    alignSelf: 'stretch',
    color: '#fff',
    padding: 20,
    borderWidth: 2,
    borderRadius: 20,
    marginBottom: '2%',
    borderColor: '#ededed',
  },
  field: {
    display: 'flex',
    flexDirection: 'row',
  },
  loginButton: {
    marginLeft: '35%',
    marginTop: 10,
    backgroundColor: '#E91E63',
    width: '25%',
    height: '20%',
    borderRadius: 20,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    elevation: 8,
  },
  loginFieldContainer: {
    width: '80%',
  },
});
