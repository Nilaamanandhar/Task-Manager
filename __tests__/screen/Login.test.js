import 'react-native';
import React from 'react';

import Login from '../../src/screen/Login';

import renderer from 'react-test-renderer';

describe('<Login />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});