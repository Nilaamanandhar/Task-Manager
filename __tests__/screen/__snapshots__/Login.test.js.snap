// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assests/image/welcome.jpg",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 40,
        "zIndex": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 25,
          "fontWeight": "500",
          "marginTop": 15,
          "opacity": 0.5,
        }
      }
    >
      log in
    </Text>
    <View
      style={
        Object {
          "width": "80%",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="User Name"
        placeholderTextColor="white"
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "stretch",
            "borderColor": "#ededed",
            "borderRadius": 20,
            "borderWidth": 2,
            "color": "#fff",
            "marginBottom": "2%",
            "padding": 20,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Password"
        placeholderTextColor="white"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "alignSelf": "stretch",
            "borderColor": "#ededed",
            "borderRadius": 20,
            "borderWidth": 2,
            "color": "#fff",
            "marginBottom": "2%",
            "padding": 20,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#E91E63",
            "borderRadius": 20,
            "display": "flex",
            "elevation": 8,
            "height": "20%",
            "justifyContent": "center",
            "marginLeft": "35%",
            "marginTop": 10,
            "width": "25%",
          }
        }
      >
        <Text>
          Login
        </Text>
      </View>
    </View>
    <View>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "marginTop": 0,
          }
        }
      >
        username: praful 
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "marginTop": 0,
          }
        }
      >
        password: praful 
      </Text>
    </View>
  </View>
</View>
`;
